### PR TITLE
confirm authenticity of id-proof keypair

### PR DESCRIPTION
**Pull request by ArtificialOwl**
store a checksum of the file-stored key in the database

### CODE REVIEW

File: lib/private/Security/IdentityProof/Exception/IdentityProofKeySumException.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+namespace OC\Security\IdentityProof\Exception;
```
Comment (nickvergessen) [inline lib/private/Security/IdentityProof/Exception/IdentityProofKeySumException.php:26]: Should be in the public namespace so it can be catched?

Comment (nickvergessen) [reply lib/private/Security/IdentityProof/Exception/IdentityProofKeySumException.php:26]: Oh wait this all has no public api?

Comment (ArtificialOwl) [reply lib/private/Security/IdentityProof/Exception/IdentityProofKeySumException.php:26]: I only did some monkeying on this one. There is no public interface on `Security\IdentityProof`. Should I keep the files as it is, or do you have an example of the structure we should expect ?

Comment (nickvergessen) [reply lib/private/Security/IdentityProof/Exception/IdentityProofKeySumException.php:26]: Let's keep as is for now and after this PR we can add a public interface for 25+

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
seems federated share acceptance tests fail now with weird 400 errors

**Comment by nickvergessen** [comment]
A rebase is recommended as the base is 3 months old

**Comment by skjnldsv** [comment]
Conflicts

**Pull request closed without merging**